pennylane/templates/tensornetworks/mera.py

```diff
@@ -227,13 +227,18 @@ def compute_decomposition(
         op_list = []
         if block.__code__.co_argcount > 2:
             for idx, w in enumerate(ind_gates):
-                op_list.append(block(*weights[idx], wires=w))
+
+                script = qml.tape.make_qscript(block)(*weights[idx], wires=w)
```

Comment:
So we will need to make sure `decomposition` acts like a quantum function and queues everything.
We could just do:
```
_ = [qml.apply(op) for op in script]
```
or something like that.

Reply:
I'm ok to make that change, but I'm wondering why we need to make sure of it? `some_op.decomposition()` is typically called in `some_op.expand()`, which [calls it](https://github.com/PennyLaneAI/pennylane/blob/436b76036be08d802b7cf60073fbc6a5f318fd67/pennylane/operation.py#L1250-L1251) within a `qml.QueuingManager.stop_recording()` context (and in a `make_qscript` call before this PR).

Reply:
Because there is an undocumented assumption that `decomposition` behaves like a qfunc, and getting rid of that assumption would be a breaking change.